atox/src/main/kotlin/ui/settings/SettingsFragment.kt

```diff
@@ -234,16 +234,22 @@ class SettingsFragment : BaseFragment<FragmentSettingsBinding>(FragmentSettingsB
             Toast.makeText(requireContext(), getString(R.string.password_updated), Toast.LENGTH_LONG).show()
         }
 
-        nospam.setText("%08X".format(vm.getNospam()))
-        nospam.doAfterTextChanged {
-            saveNospam.isEnabled =
-                nospam.text.length == 8 && nospam.text.toString().toUInt(16).toInt() != vm.getNospam()
-        }
-        saveNospam.isEnabled = false
-        saveNospam.setOnClickListener {
-            vm.setNospam(nospam.text.toString().toUInt(16).toInt())
+        if (vm.nospamAvailable()) {
+            nospam.setText("%08X".format(vm.getNospam()))
+            nospam.doAfterTextChanged {
+                saveNospam.isEnabled =
+                    nospam.text.length == 8 && nospam.text.toString().toUInt(16).toInt() != vm.getNospam()
+            }
+            saveNospam.isEnabled = false
```

Comment:
I would prefer putting this in the XML or at least calling it before the `if` statement so there's no duplicate.

Reply:
I left the duplication on purpose since these two cases have different semantic meanings and making code with different semantic meaning common makes things more complicated to think about.

The `isEnable = false` in the case where the nospam is available is because the "should the nospam be saveable" validation fails. The one in the second case is because Tox is unavailable and the nospam can never be saved.

I also don't want to move it to XML as that would spread the logic out across multiple files.

Reply:
Yeah you're right.